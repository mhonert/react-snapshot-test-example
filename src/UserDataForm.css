.UserDataForm > label {
    display: block;
    width: 10rem;
    margin-bottom: 1rem;
}

.UserDataForm > label > input {
    width: 20rem;
}
