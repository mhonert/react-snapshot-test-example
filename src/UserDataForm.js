import React from 'react';
import { useState } from 'react';

import './UserDataForm.css';

const UserDataForm = ({userData, isReadOnly, onSubmit}) => {
  const [name, changeName] = useState(userData.name);
  const [email, changeEmail] = useState(userData.email);

  const onSubmitClick = e => {
    e.preventDefault();
    onSubmit( {name, email} );
  }

  return (
    <form className="UserDataForm">
      <h2>{ isReadOnly ? "Read-only form" : "Editable form" }</h2>

      <label>Name
        <input type="text"
               readOnly={isReadOnly}
               value={name}
               onChange={e => changeName(e.target.value)} />
      </label>

      <label>E-Mail
        <input type="text"
               readOnly={isReadOnly}
               value={email}
               onChange={e => changeEmail(e.target.value)} />
      </label>

      { !isReadOnly && <button type="submit" onClick={onSubmitClick}>Submit</button> }
    </form>
  );
};

export default UserDataForm;