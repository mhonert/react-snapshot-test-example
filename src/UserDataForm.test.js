import React from 'react';
import {shallow} from 'enzyme';
import UserDataForm from "./UserDataForm";

it('Render UserDataForm in editable mode', () => {
  expect(renderUserDataForm(false)).toMatchSnapshot();
});

it('Render UserDataForm in read-only mode', () => {
  expect(renderUserDataForm(true)).toMatchSnapshot();
});

const renderUserDataForm = isReadOnly =>
  shallow(<UserDataForm isReadOnly={isReadOnly}
                        userData={{name: "Test name", email: "test@localhost"}} />);