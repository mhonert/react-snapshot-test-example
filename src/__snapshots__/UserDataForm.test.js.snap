// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UserDataForm in editable mode 1`] = `
<form
  className="UserDataForm"
>
  <h2>
    Editable form
  </h2>
  <label>
    Name
    <input
      onChange={[Function]}
      readOnly={false}
      type="text"
      value="Test name"
    />
  </label>
  <label>
    E-Mail
    <input
      onChange={[Function]}
      readOnly={false}
      type="text"
      value="test@localhost"
    />
  </label>
  <button
    onClick={[Function]}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`Render UserDataForm in read-only mode 1`] = `
<form
  className="UserDataForm"
>
  <h2>
    Read-only form
  </h2>
  <label>
    Name
    <input
      onChange={[Function]}
      readOnly={true}
      type="text"
      value="Test name"
    />
  </label>
  <label>
    E-Mail
    <input
      onChange={[Function]}
      readOnly={true}
      type="text"
      value="test@localhost"
    />
  </label>
</form>
`;
